### PR TITLE
feature(cache): allows specifying cache directory in settings.php

### LINKED ISSUE #8833: Split data dir into "uploads" and "cache" directories
Currently we have lots of directories like 1, 10000, 100000, etc. which store user-uploaded/source-of-truth content (i.e. dangerous to delete) alongside "cache" directories like system_cache, views_simplecache, etc. that store computed data (i.e. safe to delete).

It would be nice if these two kinds of filesystem data were stored in different directories so that backups only backed up the source-of-truth data instead of getting polluted with computable data.

Also, I want to have a "backups" directory for doing DB dumps and data-dir zips, but backups shouldn't include the backups directory itself. It would make it easier/cleaner to implement such a thing if the data dir looked liked:

```
dataroot/
  backups/
  cache/
  uploads/
```

rather than the current:

```
dataroot/
  1/
  10000/
  100000/
  ...
  views_simplecache/
  system_cache/
```

Still not sure if "uploads" is the right word, since I could imagine the data directory being where we store the SQLite DB file if Elgg ever supported SQLite.

### PR DESCRIPTION
**Pull request by mrclay**
All file caches (system/simple cache, views) are stored in the directory `$CONFIG->cacheroot`. If not given, this defaults to the data directory.

The system-wide file cache object is moved to a service.

### CODE REVIEW

File: engine/lib/configuration.php [hunk 1]
```diff
@@ -49,6 +49,18 @@ function elgg_get_data_path() {
...
 
 /**
+ * Get the cache directory path for this installation.
+ *
+ * If not set in settings.php, the data path will be returned.
+ *
+ * @return string
+ * @since 1.8.0
+ */
+function elgg_get_cache_path() {
```
Comment (mrclay) [inline engine/lib/configuration.php]: Exposing this isn't a necessary part of this I suppose.

**Comment by hypeJunction** [comment]
LGTM

**Comment by juho-jaakkola** [comment]
To be exact this doesn't really fix #8833, which has a bit broader goal. Maybe we should think of a more profound overhaul if we want to work on the datadir?

**Comment by mrclay** [comment]
I guess I fail to see the value of a "backups" directory (considering core doesn't provide a backup strategy), or why it or the cache directory needs to be nested inside dataroot.

**Comment by jdalsem** [comment]
@mrclay if you can remove the #fixes from the commit message this can be merged

**Comment by mrclay** [comment]
Rebased, removed `Fixes`, and added dependencies to clean up SystemCache.

**Pull request merged**